core/tests/test_utils.py

```diff
@@ -3901,9 +4487,17 @@ class FailingFunction(FunctionWrapper):
     It can be set to succeed after a given number of calls.
     """
 
-    INFINITY = 'infinity'
+    INFINITY: Final = 'infinity'
 
-    def __init__(self, f, exception, num_tries_before_success):
+    # Here we use type Any because argument 'f' can accept any kind of
+    # function signature. So, to allow every function signature we used
+    # Callable[..., Any] type here.
+    def __init__(
+        self,
+        f: Callable[..., Any],
+        exception: Union[Type[BaseException], BaseException],
+        num_tries_before_success: Union[str, int]
```

Comment:
This is quite weird can we refactor this and maybe use `math.inf` for infinity?

Reply:
refactored and tests are passing